libraries/Microsoft.Bot.Builder/TurnContext.cs

```diff
@@ -301,7 +293,7 @@ Task<ResourceResponse[]> SendActivitiesThroughCallbackPipeline(int nextCallbackI
 
             async Task<ResourceResponse> ActuallyUpdateStuff()
             {
-                return await this.Adapter.UpdateActivity(this, a);
+                return await Adapter.UpdateActivity(this, a);
```

Comment:
.ConfigureAwait(false); 

Note: This applies thorughout this code. From what I see, all of them should have that.